src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/blocks/HologramProjector.java

```diff
@@ -89,14 +109,116 @@ private static void openEditor(Player p, Block projector) {
             return false;
         });
 
-        menu.addItem(1, new CustomItem(Material.CLOCK, "&7Offset: &e" + DoubleHandler.fixDouble(Double.valueOf(BlockStorage.getLocationInfo(projector.getLocation(), OFFSET_PARAMETER)) + 1.0D), "", "&rLeft Click: &7+0.1", "&rRight Click: &7-0.1"));
-        menu.addMenuClickHandler(1, (pl, slot, item, action) -> {
-            double offset = DoubleHandler.fixDouble(Double.valueOf(BlockStorage.getLocationInfo(projector.getLocation(), OFFSET_PARAMETER)) + (action.isRightClicked() ? -0.1F : 0.1F));
+        menu.addItem(9, new CustomItem(Material.CLOCK, "&cReset Position", "", "&eClick &7to reset hologram location"));
+        menu.addMenuClickHandler(9, (pl, slot, item, action) -> {
+            ArmorStand hologram = getArmorStand(projector, true);
+            Location l = new Location(projector.getWorld(), projector.getX() + 0.5, projector.getY() + 0.5, projector.getZ() + 0.5);
+            hologram.teleport(l);
+
+            BlockStorage.addBlockInfo(projector, Y_OFFSET_PARAMETER, String.valueOf(0.5));
+            BlockStorage.addBlockInfo(projector, X_OFFSET_PARAMETER, String.valueOf(0.5));
+            BlockStorage.addBlockInfo(projector, Z_OFFSET_PARAMETER, String.valueOf(0.5));
+            openEditor(pl, projector);
+            return false;
+        });
+
+        // Y Axis
+        menu.addItem(2, new CustomItem(SkullItem.fromHash("b221da4418bd3bfb42eb64d2ab429c61decb8f4bf7d4cfb77a162be3dcb0b927"),
+            "&7Vertical Offset (+): &e" + DoubleHandler.fixDouble(Double.parseDouble(BlockStorage.getLocationInfo(projector.getLocation(), Y_OFFSET_PARAMETER))),
+            "", "&eLeft Click: &7+0.1", "&eRight Click: &7+0.5"));
+        menu.addMenuClickHandler(2, (pl, slot, item, action) -> {
+            double yOffset = DoubleHandler.fixDouble(Double.parseDouble(BlockStorage.getLocationInfo(projector.getLocation(), Y_OFFSET_PARAMETER)) + (action.isRightClicked() ? 0.5F : 0.1F));
+            double xOffset = Double.parseDouble(BlockStorage.getLocationInfo(projector.getLocation(), X_OFFSET_PARAMETER));
+            double zOffset = Double.parseDouble(BlockStorage.getLocationInfo(projector.getLocation(), Z_OFFSET_PARAMETER));
+            ArmorStand hologram = getArmorStand(projector, true);
+            Location l = new Location(projector.getWorld(), projector.getX() + xOffset, projector.getY() + yOffset, projector.getZ() + zOffset);
+            hologram.teleport(l);
+
+            BlockStorage.addBlockInfo(projector, Y_OFFSET_PARAMETER, String.valueOf(yOffset));
```

Comment:
Those are waaaaaaaaaaaayyyy too many reads/writes to BlockStorage, get your data handling sorted.